zeppelin-server/pom.xml

```diff
@@ -114,6 +114,27 @@
     </dependency>
 
     <dependency>
+      <groupId>io.buji</groupId>
+      <artifactId>buji-pac4j</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>org.pac4j</groupId>
+      <artifactId>pac4j-core</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>org.pac4j</groupId>
+      <artifactId>pac4j-oidc</artifactId>
+      <exclusions>
+        <exclusion>
+          <groupId>org.apache.commons</groupId>
+          <artifactId>commons-lang3</artifactId>
+        </exclusion>
+      </exclusions>
+    </dependency>
```

Comment:
You'll need to list license of those dependencies in `zeppelin-distribution/src/bin-license/LICENSE` file.
`licenses/LICENSE-pac4j-1.9.8` also need to be moved to `zeppelin-distribution/src/bin-license/licenses` while those dependency will be included in binary package only.

pac4j-oidc brings some transitive dependencies. License of those dependencies also need to be taken care.

```
[INFO] +- org.pac4j:pac4j-oidc:jar:1.9.8:compile
[INFO] |  +- com.nimbusds:oauth2-oidc-sdk:jar:5.24.2:compile
[INFO] |  |  +- javax.mail:mail:jar:1.4.7:compile
[INFO] |  |  |  \- javax.activation:activation:jar:1.1:compile
[INFO] |  |  +- com.github.stephenc.jcip:jcip-annotations:jar:1.0-1:compile
[INFO] |  |  +- org.apache.commons:commons-collections4:jar:4.1:compile
[INFO] |  |  +- net.minidev:json-smart:jar:1.3.1:compile
[INFO] |  |  \- com.nimbusds:lang-tag:jar:1.4.3:compile (version selected from constraint [1.4.3,))
[INFO] |  \- com.nimbusds:nimbus-jose-jwt:jar:4.26:compile
```